src/crafting.cpp

```diff
@@ -231,7 +263,8 @@ void player::craft()
     int batch_size = 0;
     const recipe *rec = select_crafting_recipe( batch_size );
     if (rec) {
-        if ( crafting_allowed() ) {
+        if ( crafting_allowed( *rec ) )
```

Comment:
Either figure out a way to astyle or internalize the rules. I'm not merging
unstyled code, and I'm not cleaning the PRs up any more.